### PR TITLE
[GLUTEN-4151][VL] initialize local&hdfs file write sink with FileSink::create function

### PR DESCRIPTION
**Pull request by dcoliversun**
## What changes were proposed in this pull request?

The WriteFileSink is not the API Velox supposed to expose, we need to initialize local&hdfs file write sink with FileSink::create function.

(Fixes: \#4151)

## How was this patch tested?

Original unit test in GA.

### CODE REVIEW

**Comment by dcoliversun** [comment]
cc @yma11 It would be nice if you have time to review this

File: cpp/velox/operators/writer/VeloxParquetDatasource.cc [hunk 1]
```diff
@@ -45,9 +45,7 @@ const int32_t kGzipWindowBits4k = 12;
 
 void VeloxParquetDatasource::init(const std::unordered_map<std::string, std::string>& sparkConfs) {
   if (strncmp(filePath_.c_str(), "file:", 5) == 0) {
-    auto path = filePath_.substr(5);
-    auto localWriteFile = std::make_unique<LocalWriteFile>(path, true, false);
-    sink_ = std::make_unique<WriteFileSink>(std::move(localWriteFile), path);
+    sink_ = dwio::common::FileSink::create(filePath_, {.pool = pool_.get()});
```
Comment (yma11) [inline cpp/velox/operators/writer/VeloxParquetDatasource.cc:48]: `pool` should not be needed for `localFileSink` and `hdfsFileSink`.  You can take a check at velox layer.

Comment (dcoliversun) [reply cpp/velox/operators/writer/VeloxParquetDatasource.cc:48]: Correct me if I'm wrong. I refer [velox code](https://github.com/facebookincubator/velox/blob/c5bcc33e93678e5bf0a7479ba16212035008a302/velox/dwio/common/tests/LocalFileSinkTest.cpp#L41-L50) to initialize the sink, could you please share velox layer?

Comment (yma11) [reply cpp/velox/operators/writer/VeloxParquetDatasource.cc:48]: Yes. Seems `pool` is always provided in velox's usage even it's not actually used.

**Review by yma11 — APPROVED** [review]

**Pull request merged**